match/lib/match/runner.rb

```diff
@@ -85,23 +104,23 @@ def run(params)
       UI.error("To do so, just pass `readonly: true` to your match call.")
       raise ex
     ensure
-      GitHelper.clear_changes
+      storage.clear_changes if storage
     end
 
-    def fetch_certificate(params: nil)
+    def fetch_certificate(params: nil, working_directory: nil)
       cert_type = Match.cert_type_sym(params[:type])
 
-      certs = Dir[File.join(params[:workspace], "certs", cert_type.to_s, "*.cer")]
-      keys = Dir[File.join(params[:workspace], "certs", cert_type.to_s, "*.p12")]
+      certs = Dir[File.join(working_directory, "certs", cert_type.to_s, "*.cer")]
+      keys = Dir[File.join(working_directory, "certs", cert_type.to_s, "*.p12")]
 
       if certs.count == 0 || keys.count == 0
         UI.important("Couldn't find a valid code signing identity in the git repo for #{cert_type}... creating one for you now")
         UI.crash!("No code signing identity found and can not create a new one because you enabled `readonly`") if params[:readonly]
-        cert_path = Generator.generate_certificate(params, cert_type)
+        cert_path = Generator.generate_certificate(params, cert_type, working_directory)
         private_key_path = cert_path.gsub(".cer", ".p12")
 
-        self.files_to_commmit << cert_path
-        self.files_to_commmit << private_key_path
+        self.files_to_commit << cert_path
+        self.files_to_commit << private_key_path
```

Comment:
This can be better kept track of in the storage, e.g. replace this with:

```
self.storage.updated_file(cert_path)
self.storage.updated_file(private_key_path)
```

That'll allow you to get rid of `files_to_commit` here completely, which is very git-centric

For some storage interfaces there'd be no concept of committing files, so for them this would be a noop.